helpers/remoteoauth/tokenoptions.go

```diff
@@ -0,0 +1,85 @@
+package remoteoauth
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/oauth2"
+)
+
+type TokenOption func(*Token)
+
+func WithAccessToken(token, tokenType string, expiry time.Time) TokenOption {
+	return func(t *Token) {
+		t.currentToken = oauth2.Token{
+			AccessToken: token,
+			TokenType:   tokenType,
+			Expiry:      expiry,
+		}
+	}
+}
+
+func (t *Token) initCloudOpts() error {
+	_, t.cloudEnabled = os.LookupEnv("CQ_CLOUD")
+	if !t.cloudEnabled {
+		return nil
+	}
+
+	t.apiToken = os.Getenv("CLOUDQUERY_API_KEY")
```

Comment:
Can we do `errors.join` and report all possible errors and not just the first one?

Maybe also put all the validation code in a dedicated function (non blocking)

Reply:
Errors joined in https://github.com/cloudquery/plugin-sdk/pull/1866/commits/6613af3c564b6f1c38d90fed7a4b64af0bf7406d

`initCloudOpts` is already simple enough, I don't think it's beneficial to separate it even more.